In [2]:
import numpy as np
import pandas as pd
student_mat = pd.read_csv('Downloads/student-mat.csv')
student_mat

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

In [28]:
student_mat['G1'].mean()

10.90886075949367

In [29]:
data_for_weighted_mean = student_mat['G1'][:4]
weight = np.array([1,2,3,4])
np.average(data_for_weighted_mean,weights=weight)

9.6

In [30]:
filter_data = student_mat['G1'] > 0
student_mat['G1'][filter_data].values

array([ 5,  5,  7, 15,  6, 15, 12,  6, 16, 14, 10, 10, 14, 10, 14, 14, 13,
        8,  6,  8, 13, 12, 15, 13, 10,  6, 12, 15, 11, 10,  9, 17, 17,  8,
       12,  8, 15, 15, 12, 14,  7, 12, 19,  8, 10,  8, 11, 19, 15,  7, 12,
       11, 11,  8, 10,  8, 14, 14,  9, 15, 10, 10,  8, 10, 10, 16, 13,  7,
        8, 16, 13, 10,  8, 12, 11,  9, 11, 11,  8,  5, 10, 11,  7, 15,  9,
        7,  8, 13, 11,  8,  7, 16,  7, 11, 11,  7, 11,  8, 11,  7,  7, 16,
       10,  7, 16, 10,  7, 16, 10, 14, 18,  7, 10, 18,  9, 15, 11, 13,  9,
       14, 16, 16, 13, 14,  8, 13,  7,  7,  7, 18, 12,  8, 10, 12,  9, 11,
       10,  4, 14, 16,  7,  9,  9, 14,  5,  8,  6, 10,  7,  8,  6, 12, 10,
        5, 11, 11, 16,  9, 17, 10,  7,  5,  7, 10,  5, 12, 10, 14,  6, 14,
        6, 13, 13,  8, 10, 10, 13,  6, 10, 10,  9, 12, 16,  9, 12, 12, 11,
       15,  8,  8, 11,  8,  7,  8, 13, 14, 17,  9, 18,  9, 16,  8,  9,  7,
       10, 10,  7, 11,  9,  7,  8, 12, 12,  6,  8, 14,  6,  6,  7,  9,  6,
        6, 16, 12, 13,  9

In [31]:
import statistics
statistics.harmonic_mean(student_mat['G1'][filter_data])

9.843044709514144

In [32]:
import scipy.stats as stats
stats.gmean(student_mat['G1'][filter_data])

10.38574828731929

In [33]:
student_mat['G1'].median()

11.0

In [34]:
student_mat['G1'].mode()

0    10
dtype: int64

In [35]:
statistics.variance(student_mat['G1'])

11.017053267364904

In [36]:
statistics.variance(student_mat['G1'])**0.5

3.319194671507669

In [37]:
student_mat['G1'].skew()

0.24061324337096948

In [38]:
student_mat['G1'].quantile([0.25,0.5,0.75])

0.25     8.0
0.50    11.0
0.75    13.0
Name: G1, dtype: float64

In [39]:
student_mat['G1'].describe()

count    395.000000
mean      10.908861
std        3.319195
min        3.000000
25%        8.000000
50%       11.000000
75%       13.000000
max       19.000000
Name: G1, dtype: float64

In [62]:
student_mat['G2'].describe()

count    395.000000
mean      10.713924
std        3.761505
min        0.000000
25%        9.000000
50%       11.000000
75%       13.000000
max       19.000000
Name: G2, dtype: float64

In [63]:
student_mat['G3'].describe()

count    395.000000
mean      10.415190
std        4.581443
min        0.000000
25%        8.000000
50%       11.000000
75%       14.000000
max       20.000000
Name: G3, dtype: float64

In [44]:
summary_student_mat = student_mat['G1'].describe()
summary_student_mat['count']

395.0

In [45]:
summary_student_mat['mean']

10.90886075949367

In [49]:
summary_student_mat['std']

3.3191946715076686

In [50]:
summary_student_mat['min']

3.0

In [51]:
summary_student_mat['25%']

8.0

In [52]:
summary_student_mat['50%']

11.0

In [53]:
summary_student_mat['75%']

13.0

In [54]:
summary_student_mat['max']

19.0

Kesimpulan :
    1. G1 memiliki 395 data student yang mengikuti ujian pada pelajaran pertama
    2. Rata - rata nilai dari G1 adalah 10.9 dimana nilai rata - rata belum terbilang maksimal karena masih setengah dari nilai maksimalnya yaitu 19
    3. Standar Deviasi dari G1 adalah 3.3 dimana hal ini standar deviasi terhtiung luas
    4. Nilai minimum pada G1 adalah 3.0 dapat ditarik kesimpulan bahwa masih ada siswa yang belum mengerti pelajaran pada G1
    5. Quartil pertama dari G1 memiliki nilai 8.0 dapat ditarik kesimpulan nilai rata-rata dari siswa yang belum mengerti adalah 8.0
    6. Quartil kedua dari G1 memiliki nilai 11.0 dapat ditarik kesimpulan nilai rata-rata dari siswa yang mengerti adalah 11.0
    7. Quartil ketiga dari G1 memiliki nilai 13.0 dapat ditarik kesimpulan nilai rata-rata dari siswa yang sangat mengerti adalah 13.0
    8. Nilai maksimum dari G1 adalah 19.0 dapat ditarik kesimpulan nilai rata-rata dari siswa paling mengerti memiliki nilai 19.0 atau nilai tertinggi

In [58]:
corr_G1_G2 = student_mat['G1'].corr(student_mat['G2'])
corr_G1_G2

0.8521180663648061

In [61]:
corr_G1_G3 = student_mat['G1'].corr(student_mat['G3'])
corr_G1_G3

0.801467932017414

Kesimpulan:
    1. Korelasi antara G1 dan G2 sangat tinggi yaitu pada nilai 0.85 sehingga dapat ditarik kesimpulan bahwa siswa yang mengerti pelajaran G1 memiliki bekal yang akan berguna di pelajaran G2.
    2. Sedangkan korelasi antara G1 dan G3 lebih rendah daripada kolerasi G1 dengan G2 yaitu dengan nilai 0.80 sehingga dapat ditarik kesimpulan bahwa materi pada G3 memiliki perbedaan dengan materi G1 sehingga dapat dipelajari secara otodidak.